lib/facter/simplib__mountpoints.rb

```diff
@@ -0,0 +1,143 @@
+
+    # Holder of the multi-call content
+    mount_list = {}
+
+    facter_mountpoints = Facter.value('mountpoints') || {}
+
+    # Sometimes Ruby has issues with /proc so fall back to a shell command
+    Facter::Core::Execution.execute('cat /proc/mounts 2> /dev/null', :on_fail => nil).each_line do |line|
+      line.strip!
+
+      next if line.empty? || line.match(%r{^\s+none\s+})
+
+      dev, path, fs, opts, _junk = line.split(%r{\s+})
+
+      next unless target_dirs.include?(path)
+
+      mount_list[path] = (facter_mountpoints[path] || {
+        'device' => dev,
+        'filesystem' => fs,
+        # Split on commas that are not in quotes
+        'options' => opts.gsub(%r{'|"}, '').split(%r{,(?=(?:(?:[^'"]*(?:'|")){2})*[^'"]*$)}).map(&:strip)
+      })
+    end
+
+    # Lookup table so we don't constantly lookup found UIDs and GIDs, etc...
+    known_translations = {}
+
+    # Check for bind mounts using findmnt since some systems (EL7) do not post the
+    # 'bind' keyword into the mount options any longer.
+    mount_list.each_key do |mnt|
+      mount_list[mnt]['options'] ||= []
+
+      unless mount_list[mnt]['options'].include?('bind')
+        findmnt_output = Facter::Core::Execution.execute("findmnt #{mnt}", :on_fail => nil)
+        if findmnt_output
+          mnt_source = findmnt_output.lines.last.split(%r{\s+})[1]
+
+          # We're a bind mount if this happens
+          if mnt_source.include?('[')
+            bind_source = mnt_source[%r{\[(.*)\]}, 1] # Match contents in brackets, extract first match
+
+            mount_list[mnt]['device'] = bind_source
+            mount_list[mnt]['filesystem'] = 'none'
+            mount_list[mnt]['options'] << 'bind'
+          end
+        end
+      end
+
+      # Add an 'options_hash' for easy processing
+      # rubocop:disable Style/MultilineBlockChain
+      mount_list[mnt]['options_hash'] = Hash[
+        mount_list[mnt]['options'].map { |opt|
+          # Split on options that are not in quotes
+          opt.split(%r{=(?=(?:(?:[^'"]*(?:'|")){2})*[^'"]*$)}).map(&:strip)
+        }.map do |opt_arr|
+          if opt_arr[1] && opt_arr[1] =~ %r{\A\d+\Z}
+            [opt_arr[0], opt_arr[1].to_i]
+          else
+            [opt_arr[0], opt_arr[1]]
+          end
+        end
+      ]
+      # rubocop:enable Style/MultilineBlockChain
+
+      # Helper translation material
+      if mount_list[mnt]['options_hash']['uid']
+        require 'etc'
+
+        begin
+          known_translations[:uid] ||= {}
+
+          found_uid = known_translations[:uid][mount_list[mnt]['options_hash']['uid']] || Etc.getpwuid(mount_list[mnt]['options_hash']['uid']).name
+          mount_list[mnt]['options_hash']['_uid__user'] = found_uid
+
+          known_translations[:uid][mount_list[mnt]['options_hash']['uid']] = found_uid
+        rescue ArgumentError
+          # noop
+        end
+      end
+
+      next unless mount_list[mnt]['options_hash']['gid']
+
```

Comment:
Is there ever a case in which the `options_hash `would have the gid but not the uid?  If so, the `require 'etc'` would need to be in the gid-handling block, as well.

Reply:
Ah, I think this broke when I `rubocop'd` things